### ISSUE TITLE
Support generation of Generic Models

### ISSUE THREAD

**Issue by rfscholte** [codehaus-plexus/modello#21]
The Maven project has several steps to come to a resolved Model (inheritance, interpolation) where values can be symbolic (ie "${something}"). Right now it means that everything has to be a String, even if the resolved value will be an integer or a boolean.
It would be an improvement if there was something like

    class Model<BLN,BT,CHR,DBL,INT,FLT,LNG,SHRT>
  
    rawModel = new Model<String,String,String,String,String,String,String>()
    resolvedModel = new Model<Byte,Boolean,Character,Double,Integer,Float,Long,Short>()

This way we don't have to do any concessions and the actual Type in the end becomes clear.

**Comment by michael-o** [comment]
Is this about the port inpolation issue in `settings.xml`?

**Comment by rfscholte** [comment]
Yes and no, but it triggered it. There's no rawSettings in Maven, because most values are just Strings and there's close to no filtering required. The port was the first one and exposed the issue, so maybe there should be a rawSettings.
The MavenProject holds the model and originalModel, and to support interpolation you'll loose the intended type. The concept above should fix that issue, no manual adjustments required.

**Comment by hboutemy** [comment]
notice that "the actual Type in the end becomes clear" will be true at use time, but not at generation time (since generics will hide the target type)
I don't know if something can be done to improve this aspect, but at least this generics idea looks promising
